### PR TITLE
feat: TLOAD and TSTORE

### PR DESCRIPTION
**Pull request by obatirou**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #846 
Resolves: #847 

## What is the new behavior?

TLOAD and TSTORE are implemented

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-ssj/873)

### CODE REVIEW

File: crates/evm/src/state.cairo [hunk 1]
```diff
@@ -184,12 +188,29 @@ impl StateImpl of StateTrait {
...
+        let maybe_entry = self.transient_account_storage.read(internal_key);
+        match maybe_entry {
+            Option::Some((_, _, value)) => { return value; },
+            Option::None => { return 0; }
+        }
+    }
+
+    #[inline(always)]
+    fn write_transient_storage(ref self: State, evm_address: EthAddress, key: u256, value: u256) {
+        let internal_key = compute_state_key(evm_address, key);
```
Comment (obatirou) [inline crates/evm/src/state.cairo:203]: What do you think about renaming `compute_state_key` to `compute_storage_key` ?
Seems better suited

Comment (enitrat) [reply crates/evm/src/state.cairo:203]: yes

File: crates/evm/src/instructions/memory_operations.cairo [hunk 2]
```diff
@@ -250,6 +250,33 @@ impl MemoryOperation of MemoryOperationTrait {
...
+    /// Save a word to the transient storage.
+    /// # Specification: https://www.evm.codes/#5d?fork=cancun
+    fn exec_tstore(ref self: VM) -> Result<(), EVMError> {
+        ensure(!self.message().read_only, EVMError::WriteInStaticContext)?;
+        let key = self.stack.pop()?;
+        let value = self.stack.pop()?;
+
+        self.charge_gas(gas::WARM_ACCESS_COST)?;
+        self.env.state.write_transient_storage(self.message().target.evm, key, value);
+
```
Comment (enitrat) [inline crates/evm/src/instructions/memory_operations.cairo:276]:
```suggestion
        let key = self.stack.pop()?;
        let value = self.stack.pop()?;
        
        self.charge_gas(gas::WARM_ACCESS_COST)?;
        
        ensure(!self.message().read_only, EVMError::WriteInStaticContext)?;
        self.env.state.write_transient_storage(self.message().target.evm, key, value);
```

stack -> gas -> operation (including checks)

Comment (obatirou) [reply crates/evm/src/instructions/memory_operations.cairo:276]: oh right 
In that case, sstore needs to be updated as well: https://github.com/kkrt-labs/kakarot-ssj/blob/4e29cc6d002fec74c5436bd6ba106ae050ce77a3/crates/evm/src/instructions/memory_operations.cairo#L107

**Review by enitrat — APPROVED** [review]

**Pull request merged**